sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1742,13 +1742,18 @@ case class MakeTimestamp(
       val seconds = secAndNanos.toInt
       val nanos = ((secAndNanos - seconds) * NANOS_PER_SECOND).toInt
       val ldt = if (seconds == 60) {
-        LocalDateTime.of(year, month, day, hour, min, 0, nanos).plusMinutes(1)
+        if (nanos == 0) {
+          // This case of sec = 60 and nanos = 0 is supported for compatibility with PostgreSQL
+          LocalDateTime.of(year, month, day, hour, min, 0, 0).plusMinutes(1)
+        } else {
+          throw new DateTimeException("The fraction of sec must be zero. Valid range is [0, 60].")
```

Comment:
Ur, not exception. I mean `null` literally.

Reply:
Oh.. It's catched at line 1756?

Reply:
Yes. The code looks better from my point of view with the exception.

Reply:
`LocalDateTime.of` throws the `DateTimeException` for invalid inputs, and need to catch it in any case. Here we have some kind of similar case.

Reply:
Got it.